xapian-letor/ranker/listmle_ranker.cc

```diff
@@ -0,0 +1,199 @@
+/** @file listmle_ranker.cc
+ *  @brief Implementation of ListMLERanker
+ */
+/* Copyright (C) 2014 Hanxiao Sun
+ * Copyright (C) 2016 Ayush Tomar
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+/*
+ListMLE is adapted from the paper:
+Fen Xia, Tie-Yan Liu, Jue Wang, Wensheng Zhang, Hang Li
+et al. "Listwise Approach to Learning to Rank - Theory and Algorithm."
+*/
+#include <config.h>
+
+#include "xapian-letor/ranker.h"
+
+#include "debuglog.h"
+#include "serialise-double.h"
+
+#include <xapian.h>
+#include <xapian/intrusive_ptr.h>
+#include <xapian/types.h>
+#include <xapian/visibility.h>
+
+#include <algorithm>
+#include <cmath>
+#include <stdlib.h>
+
+using namespace std;
+using namespace Xapian;
+
+ListMLERanker::~ListMLERanker() {
+    LOGCALL_DTOR(API, "ListMLERanker");
+}
+
+static bool labelComparer(const FeatureVector & firstfv,
```

Comment:
Xapian coding style is that functions and methods should be lower case with `_` between words, not "camel case" like here (only class names use camel case).

There seems to be a lot of instances of this problem here.